registration/tests/enhanced_backend.py

```diff
@@ -0,0 +1,258 @@
+import datetime
```

Comment:
I think that setting a class variable would be best since most of the testing logic is identical (sorry I missed the import where you imported `SupervisedRegistrationProfile` as `RegistrationProfile`)